### PR TITLE
Add IPv6 support to netmask and gateway fields

### PR DESCRIPTION
**Pull request by Nambrok**

### CODE REVIEW

**Comment by Nambrok** [comment]
@benjamreis @stormi  In need of review

File: XSConsoleData.py [hunk 1]
```diff
@@ -980,21 +980,58 @@ def ManagementIP(self, inDefault = None):
...
+                ipre = r'[0-9a-f.:]+'
+                ifRE = re.compile(r'\s*inet\s+' + ipre + '\s+netmask\s+(' + ipre + r')\s+broadcast\s+(' + ipre + r')\s*$',
+                        re.IGNORECASE)
+
+                ifconfig = commands.getoutput("/sbin/ifconfig '"+device+"'").split("\n")
+                for line in ifconfig:
+                    match = ifRE.match(line)
+                    if match:
+                        retVal = match.group(1)
+                        break
```
Comment (benjamreis) [inline XSConsoleData.py]: This code is no longer relevant Whatever the mode the netmask should be read from xapi.

File: XSConsoleData.py [hunk 2]
```diff
@@ -980,21 +980,58 @@ def ManagementIP(self, inDefault = None):
...
+                    device = pif['device']
+                routeRE = re.compile(r'([0-9.]+)\s+([0-9.]+)\s+([0-9.]+)\s+UG\s+\d+\s+\d+\s+\d+\s+'+device,
+                    re.IGNORECASE)
+
+                routes = commands.getoutput("/sbin/route -n").split("\n")
+                for line in routes:
+                    match = routeRE.match(line)
+                    if match:
+                        retVal = match.group(2)
+                        break
```
Comment (benjamreis) [inline XSConsoleData.py]: Same as above.

**Review by benjamreis — CHANGES_REQUESTED** [review]
You added back an old logic, you only need to check the right field in XAPI depending on IPv6 or not.

File: XSConsoleData.py [hunk 3]
```diff
@@ -980,22 +980,24 @@ def ManagementIP(self, inDefault = None):
 
     def ManagementNetmask(self, inDefault = None):
         retVal = inDefault
-
```
Comment (benjamreis) [inline XSConsoleData.py:983]: unecessary deletion

File: XSConsoleData.py [hunk 4]
```diff
@@ -980,22 +980,24 @@ def ManagementIP(self, inDefault = None):
...
         for pif in self.derived.managementpifs([]):
-            retVal = pif['netmask']
+            ipv6 = pif['primary_address_type'].lower() == 'ipv6'
+            try:
+                retVal = pif['IPv6'][0].split('/')[1] if ipv6 else pif['netmask']
+            except IndexError:
+                return ''
             if retVal:
                 break
-
```
Comment (benjamreis) [inline XSConsoleData.py:988]: same

File: XSConsoleData.py [hunk 5]
```diff
@@ -980,22 +980,24 @@ def ManagementIP(self, inDefault = None):
...
     def ManagementGateway(self, inDefault = None):
         retVal = inDefault
 
         for pif in self.derived.managementpifs([]):
-            retVal = pif['gateway']
+            ipv6 = pif['primary_address_type'].lower() == 'ipv6'
+            retVal = pif['ipv6_gateway'] if ipv6 else pif['gateway']
             if retVal:
                 break
-
```
Comment (benjamreis) [inline XSConsoleData.py:998]: same

**Review by benjamreis — CHANGES_REQUESTED** [review]
Since we maintain patches we try to limit stylistic changes AMAP

**Review by benjamreis — APPROVED** [review]

**Comment by benjamreis** [comment]
Upstream contribution included in https://github.com/xapi-project/xsconsole/pull/4

**Pull request merged**